### PR TITLE
Explode dependencies into individual entries

### PR DESCRIPTION
**Pull request by mulkieran**
Closes https://github.com/stratis-storage/stratisd/issues/3105

### CODE REVIEW

**Comment by mulkieran** [comment]
Still builds w/ related ci modification in https://github.com/stratis-storage/ci/pull/360

File: src/stratis/run.rs [hunk 1]
```diff
@@ -90,11 +94,15 @@ pub fn run(sim: bool) -> StratisResult<()> {
     runtime.block_on(async move {
         async fn start_threads<E>(engine: Arc<E>, sim: bool) -> StratisResult<()> where E: 'static + Engine {
             let (trigger, should_exit) = channel(1);
+
+            #[allow(unused_variables)]
```
Comment (jbaublitz) [inline src/stratis/run.rs:98]: should we `#[cfg()]` flag this instead?

**Review by jbaublitz — CHANGES_REQUESTED** [review]
Just one comment, other than that this looks good.

File: src/stratis/run.rs (hunk 1 above)
Comment (mulkieran) [reply src/stratis/run.rs]: Done. (The parser won't allow a flag inside a destructuring let, so it's a bit verbose, but still better, I think.)

**Comment by mulkieran** [comment]
also rebased.

File: src/stratis/run.rs [hunk 2]
```diff
@@ -90,11 +94,18 @@ pub fn run(sim: bool) -> StratisResult<()> {
     runtime.block_on(async move {
         async fn start_threads<E>(engine: Arc<E>, sim: bool) -> StratisResult<()> where E: 'static + Engine {
             let (trigger, should_exit) = channel(1);
+
+            #[cfg(any(feature = "dbus_enabled", feature = "min"))]
             let (udev_sender, udev_receiver) = unbounded_channel::<UdevEngineEvent>();
+            #[cfg(all(not(feature = "dbus_enabled"), not(feature = "min")))]
+            let (udev_sender, _) = unbounded_channel::<UdevEngineEvent>();
```
Comment (jbaublitz) [inline src/stratis/run.rs:101]: Both of the last two approaches seems to cause problems. The last one would generate a channel that would progressively take up more memory and never be drained and this one will generate a channel disconnect error every time a udev event is received. Neither seems ideal. I would recommend either requiring an IPC mechanism for the time being or having a separate thread to trigger udev event handling in the engine even if there is no IPC mechanism available.

**Review by jbaublitz — CHANGES_REQUESTED** [review]
I think this may require larger changes after looking more closely.

**Comment by mulkieran** [comment]
Obsoleted by #3379

**Pull request closed without merging**